tests/kani/Intrinsics/Math/Rounding/Trunc/truncf32.rs

```diff
@@ -0,0 +1,16 @@
+// Copyright Kani Contributors
+// SPDX-License-Identifier: Apache-2.0 OR MIT
+#![feature(core_intrinsics)]
+
+// Checks that `truncf32` returns the integer part of a number.
+#[kani::proof]
+fn main() {
+    let x: f32 = kani::any();
+    kani::assume(x.is_finite());
+
+    let trunc_res = unsafe { std::intrinsics::truncf32(x) };
+
+    // The expected result is the number minus its fractional part
+    let expected_res = x - x.fract();
```

Comment:
Does `fract` use `trunc` under the hood?

Reply:
Yes.

Reply:
Is that a problem for the test, if it's just running `(x - f) == (x - f)`?

Reply:
Yes, it's probably better to avoid it altogether.

It's quite hard to test these functions with symbolic values. Because of that, I've broken the `truncf32` test into a three harnesses: One that tests a property for all values and two other for concrete values.

Please let me know what you think! Any suggestions on other tests are welcome.